### PR TITLE
processor(ticdc): fix sink stats data race

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/7383

### What is changed and how it works?

Fix data race by atomic load.

#### Tests 

 - Unit test

##### Will it cause performance regression or break compatibility?

No.

##### Do you need to update user documentation, design documentation or monitoring documentation?

No.

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by 3AceShowHand — APPROVED** [review]

File: cdc/processor/pipeline/sink.go [hunk 1]
```diff
@@ -435,7 +435,7 @@ func (n *sinkNode) Stats() Stats {
 	return Stats{
 		CheckpointTs: n.CheckpointTs(),
 		ResolvedTs:   n.getResolvedTs().Ts,
-		BarrierTs:    n.barrierTs,
+		BarrierTs:    atomic.LoadUint64(&n.barrierTs),
```
Comment (hi-rustin) [inline cdc/processor/pipeline/sink.go:438]: Can we use BarrierTs()?

**Review by hi-rustin — APPROVED** [review]
Approve in advance.

**Pull request merged**